pkg/extender/extender.go

```diff
@@ -827,3 +912,176 @@ func (e *Extender) processCSIExtPodPrioritizeRequest(
+	}
+
+	if liveVMMigrationInProgress {
+		storklog.PodLog(pod).Debugf("Live VM migration in progress: (vmiName: %v, podBeingLiveMigrated: %v)", vmiName, podBeingLiveMigrated.Name)
+	} else {
+		storklog.PodLog(pod).Debugf("No ongoing live VM migration in progress. vmiName: %v", vmiName)
+	}
+
+	vmi, err := kvd.Instance().GetVirtualMachineInstance(context.TODO(), pod.Namespace, vmiName)
+	if err != nil {
+		return fmt.Errorf("unable to find VMI Info: %v", err)
+	}
+	storklog.PodLog(pod).Debugf("VMI is using PVC %v: ", vmi.RootDiskPVC)
+
+	pvId, err := e.Driver.GetVolumeIDFromPVC(vmi.RootDiskPVC, pod.Namespace)
+	if err != nil {
+		return fmt.Errorf("unable to inspect PVC: %v err: %v", vmi.RootDiskPVC, err)
+	}
+	storklog.PodLog(pod).Debugf("Volume id for scoring: %v", pvId)
+
+	volInfo, err := e.Driver.InspectVolume(pvId)
+	if err != nil {
+		return fmt.Errorf("unable to inspect volume id: %v. err: %v", pvId, err)
+	}
+
+	if liveVMMigrationInProgress {
+		podBeingLiveMigratedIsBindMounted, err := e.isUsingBindMount(podBeingLiveMigrated, volInfo)
+		if err != nil {
+			// Live migration in progress default is antihyperconvergence
+			storklog.PodLog(pod).Debugf("Unable to make determination if other pod is using bind mount. Using Antihyperconvergence")
+			e.updateVirtLauncherPodPrioritizeScores(encoder,
+				args,
+				volInfo,
+				false, /*preferLocalAttachment*/
+				true /*antihyperconvergence*/)
+			return nil
+		}
+		storklog.PodLog(pod).Debugf("Pod %v is running with same VMI owner reference and using localAttachment: %v", podBeingLiveMigrated.Name, podBeingLiveMigratedIsBindMounted)
+
+		if podBeingLiveMigratedIsBindMounted {
+			// Antihyperconvergence
+			e.updateVirtLauncherPodPrioritizeScores(encoder,
+				args,
+				volInfo,
+				false, /*preferLocalAttachment*/
+				true /*antihyperconvergence*/)
+		} else {
+			// This means existing pod is using NFS mount.
+			// Use Antihyperconvergence and give high score to volume with
+			// attached node as an exception
+			e.updateVirtLauncherPodPrioritizeScores(encoder,
+				args,
+				volInfo,
+				true, /*preferLocalAttachment*/
+				true /*antihyperconvergence*/)
+		}
+	} else {
+		storklog.PodLog(pod).Debugf("No live VM migration in progress")
+		// Default behavior for live migration not in progress is hyperconvergence
+		// We should still give a higher score to local node. We might not have
+		// attachement info avilable in this scenario and the replica score will be
+		// valid for all the replica nodes. However, if volume attachment info is available
+		// that node should get higher score.
+		e.updateVirtLauncherPodPrioritizeScores(encoder,
+			args,
+			volInfo,
+			true, /*preferLocalAttachment*/
+			false /*antihyperconvergence*/)
+	}
+	return nil
+}
+
+func (e *Extender) updateVirtLauncherPodPrioritizeScores(
+	encoder *json.Encoder,
+	args schedulerapi.ExtenderArgs,
+	volInfo *volume.Info,
+	preferLocalAttachment bool,
+	antihyperconvergence bool) {
```

Comment:
I am wondering why we need 2 params instead of just one. If `antihyperconvergence=false`, we should always pick the node where the volume is attached or a replica node if the volume is detached.

Reply:
I have answered this question in above comments too. I wil update comments in the code to make it clearer.

Reply:
Just to clarify, Antihyperconvergence since we will not chose any other volume replicas but only local attachment, which is being enforced with preferLocalAttachment flag.